app/account/orders/[id]/page.tsx

```diff
@@ -0,0 +1,288 @@
+	params,
+	searchParams,
+}: {
+	params: { id: string };
+	searchParams: { key: string };
+}) {
+	const orderToken = searchParams.key;
+	const orderId = `gid://shopify/Order/${params.id}?key=${orderToken}`;
+
+	const response = await getCustomerOrder(orderId);
+	const order = response.body.data.node;
+
+	if (!order) {
+		throw new Response('Order not found', { status: 404 });
+	}
+
+	const lineItems = flattenConnection(order.lineItems!) as Array<OrderLineItem>;
+
+	const discountApplications = flattenConnection(
+		order.discountApplications as DiscountApplicationConnection
+	);
+
+	const firstDiscount = discountApplications[0]?.value;
+
+	const discountValue =
+		firstDiscount?.__typename === 'MoneyV2' && firstDiscount;
+
+	const discountPercentage =
+		firstDiscount?.__typename === 'PricingPercentageValue' &&
+		firstDiscount?.percentage;
+
+	return (
+		<div>
+			<PageHeader heading="Order detail">
+				<Link href="/account">
+					<Text color="subtle">Return to Account Overview</Text>
+				</Link>
+			</PageHeader>
+			<div className="w-full p-6 sm:grid-cols-1 md:p-8 lg:p-12 lg:py-6">
+				<div>
+					<Text as="h3" size="lead">
+						Order No. {order.name}
+					</Text>
+					<Text className="mt-2" as="p">
+						Placed on {new Date(order.processedAt!).toDateString()}
+					</Text>
+					<div className="grid items-start gap-12 sm:grid-cols-1 md:grid-cols-4 md:gap-16 sm:divide-y sm:divide-gray-200">
+						<table className="min-w-full my-8 divide-y divide-gray-300 md:col-span-3">
+							<thead>
+								<tr className="align-baseline ">
+									<th
+										scope="col"
+										className="pb-4 pl-0 pr-3 font-semibold text-left"
+									>
+										Product
+									</th>
+									<th
+										scope="col"
+										className="hidden px-4 pb-4 font-semibold text-right sm:table-cell md:table-cell"
+									>
+										Price
+									</th>
+									<th
+										scope="col"
+										className="hidden px-4 pb-4 font-semibold text-right sm:table-cell md:table-cell"
+									>
+										Quantity
+									</th>
+									<th
+										scope="col"
+										className="px-4 pb-4 font-semibold text-right"
+									>
+										Total
+									</th>
+								</tr>
+							</thead>
+							<tbody className="divide-y divide-gray-200">
+								{lineItems.map((lineItem: OrderLineItem) => (
+									<tr key={lineItem.variant!.id}>
+										<td className="w-full py-4 pl-0 pr-3 align-top sm:align-middle max-w-0 sm:w-auto sm:max-w-none">
+											<div className="flex gap-6">
+												<Link
+													href={`/products/${
+														lineItem.variant!.product!.handle
+													}`}
+												>
+													{lineItem?.variant?.image && (
+														<div className="w-24 card-image aspect-square">
+															<Image
+																width={lineItem.variant.image.width || 96}
+																height={lineItem.variant.image.height || 96}
+																src={lineItem.variant.image.url}
```

Comment:
please change into this

```suggestion
//@ts-expect-error the Image type is wrong also in the remix hydrogen project, we receive src instead of url inside lineItem.variant.image object
src={lineItem.variant.image.src}
```